src/events/multi-asset.upsert.service.ts

```diff
@@ -64,15 +60,22 @@ export class MaspTransactionsUpsertService {
       operation.block.previousBlockHash,
     );
 
-    const [maspTransactions, users] = await this.prisma.$transaction(
+    const [multiAssets, users] = await this.prisma.$transaction(
       async (prisma) => {
         let users: Map<string, User>;
-        let maspTransactions = new Array<MaspTransaction>();
-        const head = await this.maspTransactionHeadService.head();
+        let multi_assets = new Array<MultiAsset>();
+        const head = await this.multiAssetHeadService.head();
 
         if (operation.type === BlockOperation.CONNECTED) {
-          const userGraffitis = operation.transactions.map(
-            (transaction) => transaction.assetName,
+          const userGraffitis = operation.transactions.reduce<string[]>(
```

Comment:
I think it's a bit of an anti-pattern to use `reduce` and `concat` in conjunction since we should try to avoid mutation in the accumulator function. It can effectively be replaced with a `map` if we wanted to keep a functional paradigm (however that'd need a flatten since we'd be creating a 2D array).

What do you think about just moving this into 2 for loops? I think it'd be easier to read.

Reply:
Sounds good to me, probably faster too